examples/flutter_gallery/lib/demo/pan_and_zoom_demo_transform_interaction.dart

```diff
@@ -0,0 +1,469 @@
+import 'package:flutter/material.dart';
+import 'package:vector_math/vector_math_64.dart' show Vector3;
+import 'pan_and_zoom_demo_inertial_motion.dart';
+
+// This widget allows 2D transform interaction on its child. The user can drag
+// to pan, pinch to zoom and rotate, and still get back untransformed
+// coordinates for targeted tap events.
+@immutable
+// TODO(justinmc): Need a better name. The plan is to make this a full blown
+// reusable Flutter widget.
+class TransformInteraction extends StatefulWidget {
+  const TransformInteraction({
+    // The child to perform the transformations on
+    @required this.child,
+    // The desired size of the area that can receive events
+    @required this.size,
+    // The scale will be clamped to between these values
+    this.maxScale = 2.5,
+    this.minScale = 0.8,
+    // Transforms will be limited so that the viewport can not view beyond this
+    // Rect. The Rect does not rotate with the rest of the scene, so it is
+    // always aligned with the viewport.
+    this.visibleRect,
+    // Initial values for the transform can be provided
+    this.initialTranslation,
+    this.initialScale,
+    this.initialRotation,
+    // Any and all of the possible transformations can be disabled.
+    this.disableTranslation = false,
```

Comment:
No action needed (just a thought to keep in the back of your head): in general, Flutter likes composable small widgets (which can still be brought together via a utility bigger widget). There's probably synergy that makes it easier to have all these functions in the same class, but worth considering what it might look like to have a GesturePannable, GestureZoomable, GestureRotatable widget etc (that can have superclasses or mixins etc). You can get all 3 by nesting them or having a simple GestureTransformable StatelessWidget that stacks them.

Reply:
Thanks, I will give this some serious thought when I move this widget into the framework.

I actually think this sort of pattern might help solve my `reset` parameter confusion.  Having a lower level widget that takes the `transform` matrix as a parameter would make reset nicer and allow composing the sub widgets that you suggested.

Reply:
doing this in a separate PR is ok